make release fails on SEH1

**Describe the bug**
Download repo locally.  make build works but make release fails on SEH1

**Steps to reproduce**
Download repo locally.  make build works but make release fails on SEH1

**What did you expect to see?**
build/bin/linux/arm64/amazon-cloudwatch-agent.rpm
build/bin/linux/amd64/amazon-cloudwatch-agent.rpm
build/bin/linux/arm64/amazon-cloudwatch-agent.deb
build/bin/linux/amd64/amazon-cloudwatch-agent.deb
build/bin/windows/amd64/amazon-cloudwatch-agent.zip
build/bin/darwin/amd64/amazon-cloudwatch-agent.tar.gz

**What did you see instead?**
ok      github.com/aws/amazon-cloudwatch-agent/metric/distribution/regular      0.289s  coverage: 79.6% of statements
--- FAIL: TestSEH1Distribution (0.00s)
    seh1_distribution_test.go:32: 
                Error Trace:    /Users/mjh033/GitLabRepos/GitHub/amazon-cloudwatch-agent/metric/distribution/seh1/seh1_distribution_test.go:32
                Error:          Not equal: 
                                expected: map[float64]float64{20.13119624437727:1, 29.474084421392764:1, 52.21513847164702:1}
                                actual  : map[float64]float64{20.13119624437727:1, 29.47408442139276:1, 52.21513847164702:1}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -2,3 +2,3 @@
                                  (float64) 20.13119624437727: (float64) 1,
                                - (float64) 29.474084421392764: (float64) 1,
                                + (float64) 29.47408442139276: (float64) 1,
                                  (float64) 52.21513847164702: (float64) 1
                Test:           TestSEH1Distribution
    seh1_distribution_test.go:73: 
                Error Trace:    /Users/mjh033/GitLabRepos/GitHub/amazon-cloudwatch-agent/metric/distribution/seh1/seh1_distribution_test.go:73
                Error:          Not equal: 
                                expected: map[float64]float64{20.13119624437727:2, 22.144315868814992:3, 29.474084421392764:1, 52.21513847164702:1}
                                actual  : map[float64]float64{20.13119624437727:2, 22.144315868814992:3, 29.47408442139276:1, 52.21513847164702:1}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,3 +3,3 @@
                                  (float64) 22.144315868814992: (float64) 3,
                                - (float64) 29.474084421392764: (float64) 1,
                                + (float64) 29.47408442139276: (float64) 1,
                                  (float64) 52.21513847164702: (float64) 1
                Test:           TestSEH1Distribution
FAIL
        github.com/aws/amazon-cloudwatch-agent/metric/distribution/seh1 coverage: 74.6% of statements
FAIL    github.com/aws/amazon-cloudwatch-agent/metric/distribution/seh1 0.361s


**What version did you use?**
Version: 1.247358

**What config did you use?**

RPM version 4.18.1

**Environment**
OS: MacOS 13.3

**Additional context**
Add any other context about the problem here.

On an AMD host, the test passes:
```
$ go test -v -run TestSEH1Distribution ./metric/distribution/seh1/
=== RUN   TestSEH1Distribution
--- PASS: TestSEH1Distribution (0.00s)
PASS
ok      github.com/aws/amazon-cloudwatch-agent/metric/distribution/seh1 0.003s
```

On an ARM host, the test consistently fails:
```
$ go test -v -run TestSEH1Distribution ./metric/distribution/seh1/
=== RUN   TestSEH1Distribution
    seh1_distribution_test.go:32: 
                Error Trace:    /home/ec2-user/amazon-cloudwatch-agent/metric/distribution/seh1/seh1_distribution_test.go:32
                Error:          Not equal: 
                                expected: map[float64]float64{20.13119624437727:1, 29.474084421392764:1, 52.21513847164702:1}
                                actual  : map[float64]float64{20.13119624437727:1, 29.47408442139276:1, 52.21513847164702:1}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -2,3 +2,3 @@
                                  (float64) 20.13119624437727: (float64) 1,
                                - (float64) 29.474084421392764: (float64) 1,
                                + (float64) 29.47408442139276: (float64) 1,
                                  (float64) 52.21513847164702: (float64) 1
                Test:           TestSEH1Distribution
    seh1_distribution_test.go:73: 
                Error Trace:    /home/ec2-user/amazon-cloudwatch-agent/metric/distribution/seh1/seh1_distribution_test.go:73
                Error:          Not equal: 
                                expected: map[float64]float64{20.13119624437727:2, 22.144315868814992:3, 29.474084421392764:1, 52.21513847164702:1}
                                actual  : map[float64]float64{20.13119624437727:2, 22.144315868814992:3, 29.47408442139276:1, 52.21513847164702:1}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,3 +3,3 @@
                                  (float64) 22.144315868814992: (float64) 3,
                                - (float64) 29.474084421392764: (float64) 1,
                                + (float64) 29.47408442139276: (float64) 1,
                                  (float64) 52.21513847164702: (float64) 1
                Test:           TestSEH1Distribution
--- FAIL: TestSEH1Distribution (0.00s)
FAIL
FAIL    github.com/aws/amazon-cloudwatch-agent/metric/distribution/seh1 0.004s
FAIL
```

--- a/metric/distribution/seh1/seh1_distribution_test.go
+++ b/metric/distribution/seh1/seh1_distribution_test.go
@@ -4,6 +4,7 @@
 package seh1
 
 import (
+	"math/big"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -24,11 +25,12 @@ func TestSEH1Distribution(t *testing.T) {
 	assert.Equal(t, "Count", dist.Unit())
 	values, counts := dist.ValuesAndCounts()
 	assert.Equal(t, len(values), len(counts))
-	valuesCountsMap := map[float64]float64{}
+	valuesCountsMap := map[string]float64{}
 	for i := 0; i < len(values); i++ {
-		valuesCountsMap[values[i]] = counts[i]
+		valuesCountsMap[truncate(values[i])] = counts[i]
 	}
-	expectedValuesCountsMap := map[float64]float64{20.13119624437727: 1, 29.474084421392764: 1, 52.21513847164702: 1}
+	expectedValuesCountsMap := map[string]float64{"20.13119624": 1, "29.47408442": 1, "52.21513847": 1}
+
 	assert.Equal(t, expectedValuesCountsMap, valuesCountsMap)
 
 	//another dist new and add entry
@@ -45,11 +47,11 @@ func TestSEH1Distribution(t *testing.T) {
 	assert.Equal(t, "", anotherDist.Unit())
 	values, counts = anotherDist.ValuesAndCounts()
 	assert.Equal(t, len(values), len(counts))
-	valuesCountsMap = map[float64]float64{}
+	valuesCountsMap = map[string]float64{}
 	for i := 0; i < len(values); i++ {
-		valuesCountsMap[values[i]] = counts[i]
+		valuesCountsMap[truncate(values[i])] = counts[i]
 	}
-	expectedValuesCountsMap = map[float64]float64{20.13119624437727: 1, 22.144315868814992: 3}
+	expectedValuesCountsMap = map[string]float64{"20.13119624": 1, "22.14431587": 3}
 	assert.Equal(t, expectedValuesCountsMap, valuesCountsMap)
 
 	//clone dist and anotherDist
@@ -65,11 +67,11 @@ func TestSEH1Distribution(t *testing.T) {
 	assert.Equal(t, "Count", dist.Unit())
 	values, counts = dist.ValuesAndCounts()
 	assert.Equal(t, len(values), len(counts))
-	valuesCountsMap = map[float64]float64{}
+	valuesCountsMap = map[string]float64{}
 	for i := 0; i < len(values); i++ {
-		valuesCountsMap[values[i]] = counts[i]
+		valuesCountsMap[truncate(values[i])] = counts[i]
 	}
-	expectedValuesCountsMap = map[float64]float64{20.13119624437727: 2, 22.144315868814992: 3, 29.474084421392764: 1, 52.21513847164702: 1}
+	expectedValuesCountsMap = map[string]float64{"20.13119624": 2, "22.14431587": 3, "29.47408442": 1, "52.21513847": 1}
 	assert.Equal(t, expectedValuesCountsMap, valuesCountsMap)
 
 	//add distClone into another dist
@@ -90,4 +92,8 @@ func cloneSEH1Distribution(dist *SEH1Distribution) *SEH1Distribution {
 		clonedDist.buckets[k] = v
 	}
 	return clonedDist
+}
+
+func truncate(f float64) string {
+	return big.NewFloat(f).SetPrec(100).String()
 }

